src/main/java/net/minecraftforge/common/util/ITeleporter.java

```diff
@@ -58,10 +75,68 @@
     default Entity placeEntity(Entity entity, ServerWorld currentWorld, ServerWorld destWorld, float yaw, Function<Boolean, Entity> repositionEntity) {
        return repositionEntity.apply(true);
     }
-    
-    //used internally to handle vanilla hardcoding
+
+    // States if this teleporter is the vanilla instance
     default boolean isVanilla()
     {
         return getClass() == Teleporter.class;
     }
+
+    // Replicated from the vanilla code for interface implementation. Return an empty optional if no portal was found.
+    default Optional<TeleportationRepositioner.Result> findPortal(ServerWorld fromWorld, ServerWorld toWorld, Entity entity)
+    {
+        PointOfInterestManager pointofinterestmanager = toWorld.getPointOfInterestManager();
```

Comment:
Copying vanilla code should always be avoided if possible. Please check if you can't replace this with a call to the original method if it is a clean copy or patch and call the original method if the change is small enough.

Reply:
fixed in https://github.com/SilverDavidMC/MinecraftForge/pull/3